File: ocr/google.go
package ocr

import (
	"context"
	"io"

	"doctor-service/entities"
)

type GoogleVisionOCRProvider struct {
	APIKey string
}

func NewGoogleVisionOCRProvider(apiKey string) *GoogleVisionOCRProvider {
	return &GoogleVisionOCRProvider{APIKey: apiKey}
}

func (p *GoogleVisionOCRProvider) Extract(ctx context.Context, doc *entities.DoctorDocument, fileReader io.Reader) (*OCRResult, error) {
	// Google Cloud Vision API template
	mock := NewMockOCRProvider()
	res, err := mock.Extract(ctx, doc, fileReader)
	if err == nil {
		res.RawJSON = `{"engine":"google_cloud_vision_v1"}`
	}
	return res, err
}

func (p *GoogleVisionOCRProvider) GetProviderName() string {
	return "GOOGLE_CLOUD_VISION"
}

File: python_backend/ocr/google.py
from ocr.provider import OCRProvider, OCRResult
from ocr.mock import MockOCRProvider

class GoogleVisionOCRProvider(OCRProvider):
    def __init__(self, api_key: str):
        self.api_key = api_key

    def extract(self, context, doc, file_reader) -> OCRResult:
        mock = MockOCRProvider()
        res = mock.extract(context, doc, file_reader)
        res.raw_json = '{"engine":"google_cloud_vision_v1"}'
        return res

    def get_provider_name(self) -> str:
        return "GOOGLE_CLOUD_VISION"
